src/ring_theory/integrally_closed.lean

```diff
@@ -87,6 +87,14 @@ lemma is_integral_iff {x : K} : is_integral R x ↔ ∃ y, algebra_map R K y = x
 is_integral_closure.is_integral_iff
 
 omit iic
+
+lemma exists_algebra_map_eq_of_pow_mem {R K : Type*} [comm_ring R] [field K] [algebra R K]
```

Comment:
Instead of using a subalgebra, I think you can use three rings with the appropriate `is_scalar_tower`, obtaining a more general result.

Reply:
Indeed I don't understand why you need `R` at all.

Reply:
I could let `S` be a subring of `R`, but my example is an integral closure of `K`, which is a subalgebra, and subalgebras are apparently not necessarily subrings?

Reply:
I would prove something like
```
lemma exists_algebra_map_eq_of_pow_mem {R K L : Type*} [comm_ring R] [field K] [field L]
  [algebra R K] [algebra R L] [algebra K L] [is_scalar_tower R K L] [is_fraction_ring R K] 
  [is_domain R] [is_integrally_closed R] {x : L}
  {n : ℕ} (hn : 0 < n) (hx : is_integral R (x ^ n)) : is_integral R x := sorry
```
(maybe `L` can be any commutative ring, you should try to minimize the assumptions).